static/app/components/searchSyntax/parser.tsx

```diff
@@ -1173,6 +1191,10 @@ export type SearchConfig = {
    * Text filter keys we allow to have operators
    */
   textOperatorKeys: Set<string>;
+  /**
+   * When true, the parser will not parse paren groups and will return individual paren tokens
+   */
+  flattenParenGroups?: boolean;
```

Comment:
Out of those scope here, but something to think of for future work is that we could benefit from having the parsing logic not try to generate an AST. Right now the grammar does both lexing and parsing, which makes error recovery (as seen here) tricky to handle. 

We actually want the tokenization pass to succeed regardless if the syntax is valid, as it makes recovering from errors much easier (validation and AST generation is trivial once we have the list of tokens).

Reply:
Agreed. After this change there are actually very few things that can cause the parser to fail completely, but I agree separating the two would be simpler conceptually and makes things a bit easier on ourselves.